R/tfrmt_checks.R

```diff
@@ -284,3 +284,24 @@ check_footnote_plan <- function(x){
 
 }
 
+
+#' Check if the plan parameter is consistent in tfrmt
+#' @noRd
+#' @param tfrmt_object tfrmt object to be checked
+#' @param plan name of plan, e.g., "col_style_plan"
+#' @importFrom rlang abort
+#'
+#'
+check_plan <- function(tfrmt_object, plan) {
+  # extract the plan element from the tfrmt_object
+  plan_element <- tfrmt_object[[plan]]
+
+  # check if the user supplied a value to the plan parameter
+  if (!is.null(plan_element)) {
+    # Check if the plan element inherits the plan attribute
+    if (!inherits(plan_element, plan)) {
+      stop("Invalid *_plan function has been supplied to the ", plan, " parameter")
```

Comment:
Apologies for not catching this on my previous review- it would be good to use rlang::abort instead, as this will allow the error to show up as "Error in tfrmt()" rather than "Error in check_plan()". Two changes you will have to make: 
(1) pass the parent environment as another argument in the function
(2) change to rlang::abort, passing the environment into its `call` argument 

There is a bit of this happening in some of the other helpers but it might be easier to talk through it together. more info here: https://rlang.r-lib.org/reference/topic-error-call.html#passing-the-user-context